### PR TITLE
Remove -march=native from compiler flags and add instruction set support in Readme.

### PR DESCRIPTION
**Pull request by kgpai**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -93,6 +93,20 @@ Run `make` in the root directory to compile the sources. For development, use
 `make debug` to build a non-optimized debug version, or `make release` to build
 an optimized version.  Use `make unittest` to build and run tests.
 
+Note that,
+* Velox requires C++17 , thus minimum supported compiler is GCC 5.0 and Clang 5.0.
+* Velox requires the CPU to support instruction sets:
+  * bmi
```
Comment (kgpai) [inline README.md:99]: My compilation failed without bmi/bmi2 even with Jimmys latest changes, so these are still required.

File: README.md [hunk 2]
```diff
@@ -93,6 +93,20 @@ Run `make` in the root directory to compile the sources. For development, use
...
+  * bmi2
+  * f16c
+* Velox tries to use the following (or equivalent) instruction sets where available:
+  * On Intel CPUs
+    * avx  
+    * avx2
+    * sse
+  * On ARM
+    * Neon
```
Comment (kgpai) [inline README.md:108]: This is a wip I believe , but we should be supporting this shortly @Yuhta

File: README.md (hunk 1 above)
Comment (Yuhta) [reply README.md:99]: We will have some recommended flags which will include BMI2 and AVX2, and another set of minimal flags (e.g. SSE4.2).

File: CMakeLists.txt [hunk 3]
```diff
@@ -144,7 +144,7 @@ set(POSITION_INDEPENDENT_CODE True)
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 set(CMAKE_CXX_FLAGS
-    "${CMAKE_CXX_FLAGS} -mavx2 -mfma -mavx -mf16c -mbmi2 -march=native -mno-avx512f -D USE_VELOX_COMMON_BASE"
+    "${CMAKE_CXX_FLAGS} -mavx2 -mfma -mavx -mf16c -mbmi2 -mbmi -march=x86-64 -mno-avx512f -D USE_VELOX_COMMON_BASE"
```
Comment (Yuhta) [inline CMakeLists.txt:147]: If we are not using `native`, we can remove the `no-avx512f` as well.
Also `x86-64` is default on Intel and AMD so you can omit that.

**Review by Yuhta — APPROVED** [review]

**Pull request closed without merging**